### PR TITLE
opentelemetry-collector-5061 replacing ext map with host

### PR DESCRIPTION
**Pull request by shree007**
**Description:** 
Adding `component.Host` into `confighttp.HTTPClientSettings.ToClient `to follow similar norm in other places

**Link to tracking Issue:** #5061 

**Testing:**
Paasing `componenttest.NewNopHost()` into testing method

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@shree007 can you update the PR to pass the tests in this repo (not the contrib)

**Review by codeboten — COMMENTED** [review]
@shree007 any update on moving this work forward?

**Comment by shree007** [comment]
> 

Should I remove that check ?

File: config/confighttp/confighttp_test.go [hunk 1]
```diff
@@ -255,6 +249,7 @@ func TestHTTPClientSettingWithAuthConfig(t *testing.T) {
 		shouldErr    bool
 		settings     HTTPClientSettings
 		extensionMap map[config.ComponentID]component.Extension
+		host         component.Host
```
Comment (jpkrohling) [inline config/confighttp/confighttp_test.go:252]: No cases are setting these. Should it really be added here?

Comment (shree007) [reply config/confighttp/confighttp_test.go:252]: https://github.com/open-telemetry/opentelemetry-collector/pull/5083/files#diff-ae54372cbf36c6ca3ce262dc39d7db67e1d71bba20c8838a9906bc123133c8b6R312
Here its being used.

Comment (jpkrohling) [reply config/confighttp/confighttp_test.go:252]: It's consumed there, but no tests are creating a custom value for it. Each test is defining a map of components, but that's not being used anymore. Shouldn't you create a host based on that map?

Comment (shree007) [reply config/confighttp/confighttp_test.go:252]: @jpkrohling  let me walk through it again to end to end flow. I will create host based on that map accordingly.

Comment (bogdandrutu) [reply config/confighttp/confighttp_test.go:252]: It is not used anywhere.

**Review by bogdandrutu — COMMENTED** [review]
Please make PR pass tests

**Comment by shree007** [comment]
Ok. I will do today

**Pull request closed without merging**